paas_app_charmer/rabbitmq.py

```diff
@@ -0,0 +1,219 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""RabbitMQ library for handling the rabbitmq interface.
+
+The project https://github.com/openstack-charmers/charm-rabbitmq-k8s provides
+a library for the requires part of the rabbitmq interface.
+
+However, there are two charms that provide the rabbitmq interface, being incompatible:
+ - https://github.com/openstack-charmers/charm-rabbitmq-ks8 (https://charmhub.io/rabbitmq-k8s)
+ - https://github.com/openstack/charm-rabbitmq-server/ (https://charmhub.io/rabbitmq-server)
+
+The main difference is that rabbitmq-server does not publish the information in the app
+part in the relation bag. This python library unifies both charms, using a similar
+approach to the rabbitmq-k8s library.
+
+For rabbitmq-k8s, the password and hostname are required in the app databag. The full
+list of hostnames can be obtained from the ingress-address in each unit.
+
+For rabbitmq-server, the app databag is empty. The password and hostname are in the units databags,
+being the password equal in all units. Each hostname may point to different addresses. One
+of them will chosen as the in the rabbitmq parameters.
+
+rabbitmq-server support ssl client certificates, but are not implemented in this library.
+
+This library is very similar and uses the same events as
+ the library charms.rabbitmq_k8s.v0.rabbitmq.
+See https://github.com/openstack-charmers/charm-rabbitmq-k8s/blob/main/lib/charms/rabbitmq_k8s/v0/rabbitmq.py  # pylint: disable=line-too-long # noqa: W505
+"""
+
+
+import logging
+
+from ops import CharmBase, HookEvent
+from ops.framework import EventBase, EventSource, Object, ObjectEvents
+from ops.model import Relation
+
+from paas_app_charmer.charm_state import RabbitMQParameters
+
+logger = logging.getLogger(__name__)
+
+
+class RabbitMQConnectedEvent(EventBase):
+    """RabbitMQ connected Event."""
+
+
+class RabbitMQReadyEvent(EventBase):
+    """RabbitMQ ready for use Event."""
+
+
+class RabbitMQGoneAwayEvent(EventBase):
+    """RabbitMQ relation has gone-away Event."""
+
+
+class RabbitMQServerEvents(ObjectEvents):
+    """Events class for `on`.
+
+    Attributes:
+        connected: rabbitmq relation is connected
+        ready: rabbitmq relation is ready
+        goneaway: rabbitmq relation has been removed
+    """
+
+    connected = EventSource(RabbitMQConnectedEvent)
+    ready = EventSource(RabbitMQReadyEvent)
+    goneaway = EventSource(RabbitMQGoneAwayEvent)
```

Comment:
Not very convinced by this naming

Reply:
*departed* might be better

Reply:
The names used are the same ones as in the original [rabbitmq-k8s library](https://github.com/openstack-charmers/charm-rabbitmq-k8s/blob/50f5757b455cbd7ed4a3fff20d6826c6203a4868/lib/charms/rabbitmq_k8s/v0/rabbitmq.py#L118).

But sure, I have no preference, updated to departed.